spirv_hlsl.hpp

```diff
@@ -39,6 +44,8 @@ class CompilerHLSL : public CompilerGLSL
 
 		// Allows the PointSize builtin, and ignores it, as PointSize is not supported in HLSL.
 		bool point_size_compat = false;
+
+		std::vector<RootConstants> root_constants_layout = {};
```

Comment:
It's probably best to pass this as a separate function call rather than through options.

Reply:
On constructor call, overloading `compile` or do you have something else in mind?

Reply:
A separate `void set_root_constant_layouts()` is probably fine, which can be called before compile().